src/pages/playlistpage/playlistpage.js

```diff
@@ -1,22 +1,72 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import '../videolisting/videolisting.css';
 import Videocard from '../../components/videocard/videocard';
 import '../history/history.css';
+import { useParams } from 'react-router-dom';
+import { useStateContext } from '../../context/stateContext';
+import axios from 'axios';
+import { useAuth } from '../../context/authContext';
 
 export default function Playlistpage() {
-    const arr = [1, 2, 3, 4];
+    const { state, dispatch } = useStateContext();
+    const { encodedToken } = useAuth();
+    const [videos, setVideos] = useState([]);
+    const [playlistName, setPlaylistName] = useState();
+    const { _id } = useParams();
+    console.log(state);
+    useEffect(() => {
+        state.playlists.map(e => {
+            if (e._id === _id) {
+                setVideos(e.videos);
+                setPlaylistName(e.title);
+            }
+        })
```

Comment:
Here instead of using **map func** , you could have used **Array.find()** which would be more appropriate, since you need **only one playlist from the array of the playlist.**

*find() method gives the first element that satisfies the condition.*